src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -109,20 +109,36 @@ private static string NormalizePath(string path)
         /// <param name="isContainer">
         /// return true if path points to a directory else returns false.
         /// </param>
-        /// <returns></returns>
+        /// <returns>FileInfo or DirectoryInfo object.</returns>
+        /// <exception cref="System.ArgumentNullException">
+        /// path is null
+        /// </exception>
+        /// <exception cref="System.IO.IOException">
+        /// I/O error occurs
+        /// </exception>
+        /// <exception cref="System.UnauthorizedAccessException">
+        /// an I/O error or a specific type of security error
+        /// </exception>
         private static FileSystemInfo GetFileSystemInfo(string path, ref bool isContainer)
         {
-            isContainer = false;
+            // We use 'FileInfo.Attributes' (not 'FileInfo.Exist')
+            // because we want to get exceptions
+            // like UnauthorizedAccessException or IOException.
+            FileSystemInfo fsinfo = new FileInfo(path);
+            var attr = fsinfo.Attributes;
+            var exists = (int)attr != -1;
+            isContainer = exists && attr.HasFlag(FileAttributes.Directory);
```

Comment:
I see code like this for checking if a path exists and whether it's directory at 8 places. I think keeping the redundant code in `Utils.ItemExists` is better. Thoughts?

Reply:
Thanks @iSazonov